### ISSUE TITLE
sss_ssh_knownhostsproxy prevents connection if the network is unreachable via one IP address

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1502]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1498
- Created at 2012-08-23 14:26:23 by [pbrezina](https://pagure.io/user/pbrezina)
- Closed at 2017-05-03 14:00:54 as Fixed
- Assigned to nobody
---

If ssh uses sss_ssh_knownhostproxy, it cannot established connection to a hostname if an address is unreachable.

/etc/ssh/ssh_config

    GlobalKnownHostsFile2 /var/lib/sss/pubconf/known_hosts
    ProxyCommand /usr/bin/sss_ssh_knownhostsproxy -p %p %h

It seems that sss_ssh_knownhostsproxy tries only one IP address of a host. If the host is via this address unreachable, ssh fails to connect. If ssh is used without sss_ssh_knownhostsproxy, it is able to establish the connection successfully. 

    [vm-024: ~]$ ssh -vvv fedorapeople.org
    OpenSSH_5.9p1, OpenSSL 1.0.0j-fips 10 May 2012
    debug1: Reading configuration data /etc/ssh/ssh_config
    debug1: /etc/ssh/ssh_config line 50: Deprecated option "GlobalKnownHostsFile2"
    debug1: /etc/ssh/ssh_config line 53: Applying options for *
    debug2: ssh_connect: needpriv 0
    debug1: Executing proxy command: exec /usr/bin/sss_ssh_knownhostsproxy -p 22 fedorapeople.org
    debug1: permanently_drop_suid: 529
    debug3: Incorrect RSA1 identifier
    debug3: Could not load "/home/pbrezina/.ssh/id_rsa" as a RSA1 public key
    debug1: identity file /home/pbrezina/.ssh/id_rsa type 1
    debug1: identity file /home/pbrezina/.ssh/id_rsa-cert type -1
    debug1: identity file /home/pbrezina/.ssh/id_dsa type -1
    debug1: identity file /home/pbrezina/.ssh/id_dsa-cert type -1

    [vm-024: ~]$ /usr/bin/sss_ssh_knownhostsproxy -p 22 fedorapeople.org --debug 10
    (Thu Aug 23 08:12:27:146532 2012) [/usr/bin/sss_ssh_knownhostsproxy] [main] (0x0040): sss_ssh_get_ent() failed (111): Connection refused
    (Thu Aug 23 08:12:27:147323 2012) [/usr/bin/sss_ssh_knownhostsproxy] [connect_socket] (0x0040): connect() failed (101): Network is unreachable

    [vm-024: ~]$ host fedorapeople.org
    fedorapeople.org has address 152.19.134.191
    fedorapeople.org has IPv6 address 2610:28:3090:3001:5054:ff:fedb:7f5a
    [vm-024: ~]$ ssh 152.19.134.191
    (works)
    [vm-024: ~]$ ssh 2610:28:3090:3001:5054:ff:fedb:7f5a
    ssh: connect to host 2610:28:3090:3001:5054:ff:fedb:7f5a port 22: Network is unreachable

### Comments

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2012-09-06 15:19:11](https://pagure.io/SSSD/sssd/issue/1498#221482)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD Deferred
rhbz:  => todo

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2014-02-12 17:13:03](https://pagure.io/SSSD/sssd/issue/1498#221483)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=1063278 (*Fedora*)

rhbz: todo => [https://bugzilla.redhat.com/show_bug.cgi?id=1063278 1063278]

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-03-22 18:30:32](https://pagure.io/SSSD/sssd/issue/1498#221484)**

Can we bump this one in priority?  With the increased adoption of IPv6, I now see this quite often.  We have public DNS servers returning AAAA records, but then IPv6 gets blocked by network, and services fail.

changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
mark:  => 0
review:  => 0
selected:  => 
sensitive:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-03-22 20:45:13](https://pagure.io/SSSD/sssd/issue/1498#221485)**

If you're seeing issues then we should move the ticket out of deferred, but I don't think it's realistic to squeeze it into 1.14 (unless patches are provided :-)) therefore I would propose 1.15

milestone: SSSD Deferred => NEEDS_TRIAGE

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-03-29 12:26:56](https://pagure.io/SSSD/sssd/issue/1498#221486)**

This should be a 1.14 stretch goal, but we're not sure we will have the capacity to fix the bug in 1.14.0

milestone: NEEDS_TRIAGE => SSSD 1.14 backlog

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-10-20 20:22:22](https://pagure.io/SSSD/sssd/issue/1498#221487)**

Any updates on this? Could you at least add a decent error message.

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-12-06 11:44:15](https://pagure.io/SSSD/sssd/issue/1498#221488)**

Replying to [comment:6 jimmyhedman]:
> Any updates on this? Could you at least add a decent error message.
There is a candidate fix in the bugzilla linked to this message, but not all reports were positive. It would be nice if you could test the patch linked to that bugzilla.

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-12-15 11:00:47](https://pagure.io/SSSD/sssd/issue/1498#221489)**

Since the 1.14 branch is transitioning into maintenance mode and new functionality is being developed in master which will become 1.15 eventually, I'm mass-moving tickets from the 1.14 backlog milestone to the "Future releases" milestone.

milestone: SSSD 1.14 backlog => SSSD Future releases (no date set yet)

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-02-24 14:23:53](https://pagure.io/SSSD/sssd/issue/1498#221490)**

**Metadata Update from @pbrezina**:
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-04-10 21:08:58](https://pagure.io/SSSD/sssd/issue/1498#435992)**

So apparently #3366 is a duplicate of this, and it is worrying to see that we do not have a target release to fix this ticket.
Is there a documented workaround ?

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-04-28 19:18:48](https://pagure.io/SSSD/sssd/issue/1498#438881)**

master:

* 244adc327f7e29ba2c7ef60bc9f732d8fe3e68c9
* 5f6232c7e6d9635c1d6b6b09f799309b6094b143
* 08084b1179bb9fc38bc22b464b3d44907107bfd3

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-04-28 19:18:50](https://pagure.io/SSSD/sssd/issue/1498#438882)**

**Metadata Update from @lslebodn**:
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-05-03 14:00:12](https://pagure.io/SSSD/sssd/issue/1498#439251)**

sssd-1-14:

* 47c80d680f41e9ac33a10873e6d78343f14ff77a
* 078a82a38e820fbe0325a59c106263b69d2e8300
* c4fb93aa64abcb6abf0f43129483670e141f1423

sssd-1-13:

* b6e1a42ccc17a9433b15c66d7203471dd7d435a2
* 92f59195def80b9afe5fbe58f56228921d2b8c1f
* b14a9afb961a7b2dd139e7dfe6c2a5a389f3a976

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-05-03 14:00:14](https://pagure.io/SSSD/sssd/issue/1498#439252)**

**Metadata Update from @lslebodn**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-05-03 14:00:58](https://pagure.io/SSSD/sssd/issue/1498#439253)**

**Metadata Update from @lslebodn**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue close_status updated to: Fixed
- Issue set to the milestone: SSSD 1.15.3 (was: SSSD Future releases (no date set yet))
- Issue status updated to: Closed (was: Open)
